### PR TITLE
[FLINK-4096] Ensure JarOutputStream is always closed

### PR DESCRIPTION
**Pull request by mushketyk**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/util/JarFileCreator.java [hunk 1]
```diff
@@ -189,33 +189,33 @@ public synchronized void createJarFile() throws IOException {
...
+					num = classInputStream.read(buf);
+				}
 
-			classInputStream.close();
-			jos.closeEntry();
+				classInputStream.close();
+				jos.closeEntry();
+			}
 		}
-		jos.close();
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/util/JarFileCreator.java]: close() is now never called.

**Comment by mushketyk** [comment]
Since jos is now used in try-with-resource, I assume it will be called in any case:
https://github.com/apache/flink/pull/2172/files#diff-966966a476efb23e920314f9bf266539R192

Am I missing something?

**Comment by zentol** [comment]
no, you are correct.

**Comment by mushketyk** [comment]
@zentol Can you merge this PR if you find it useful?

**Pull request closed without merging**